public/app/plugins/datasource/mysql/response_parser.ts

```diff
@@ -120,47 +85,40 @@ export default class ResponseParser {
     return false;
   }
 
-  transformAnnotationResponse(options: any, data: any) {
-    const table = data.data.results[options.annotation.name].tables[0];
-
-    let timeColumnIndex = -1;
-    let timeEndColumnIndex = -1;
-    let textColumnIndex = -1;
-    let tagsColumnIndex = -1;
-
-    for (let i = 0; i < table.columns.length; i++) {
-      if (table.columns[i].text === 'time_sec' || table.columns[i].text === 'time') {
-        timeColumnIndex = i;
-      } else if (table.columns[i].text === 'timeend') {
-        timeEndColumnIndex = i;
-      } else if (table.columns[i].text === 'title') {
-        throw {
-          message: 'The title column for annotations is deprecated, now only a column named text is returned',
-        };
-      } else if (table.columns[i].text === 'text') {
-        textColumnIndex = i;
-      } else if (table.columns[i].text === 'tags') {
-        tagsColumnIndex = i;
-      }
+  async transformAnnotationResponse(options: any, data: BackendDataSourceResponse): Promise<AnnotationEvent[]> {
+    const frames = toDataQueryResponse({ data: data }).data as DataFrame[];
+    const frame = frames[0];
+    const timeField = frame.fields.find((f) => f.name === 'time' || f.name === 'time_sec');
+
+    if (!timeField) {
+      return Promise.reject({ message: 'Missing mandatory time column (with time column alias) in annotation query.' });
```

Comment:
`Promise.reject` should be the same as throwing an exception in an `async` function, you should also use `Error` objects for exceptions/rejections:
```suggestion
      throw new Error('Missing mandatory time column (with time column alias) in annotation query');
```